sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventgrid/AzureEventGridAutoConfiguration.java

```diff
@@ -0,0 +1,104 @@
+
+@Configuration(proxyBeanMethods = false)
+@EnableConfigurationProperties
+@ConditionalOnClass(EventGridPublisherClientBuilder.class)
+@ConditionalOnProperty(value = "spring.cloud.azure.eventgrid.enabled", havingValue = "true", matchIfMissing = true)
+@ConditionalOnAnyProperty(prefix = "spring.cloud.azure.eventgrid", name = "endpoint")
+public class AzureEventGridAutoConfiguration extends AzureServiceConfigurationBase {
+
+    protected AzureEventGridAutoConfiguration(AzureGlobalProperties azureProperties) {
+        super(azureProperties);
+    }
+
+    @Bean
+    @ConfigurationProperties(prefix = AzureEventGridProperties.PREFIX)
+    AzureEventGridProperties azureEventGridProperties() {
+        return loadProperties(getAzureGlobalProperties(), new AzureEventGridProperties());
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnProperty(value = "spring.cloud.azure.eventgrid.event-schema", havingValue = "EVENT_GRID_EVENT", matchIfMissing = true)
+    EventGridPublisherClient<EventGridEvent> eventGridEventPublisherClient(EventGridPublisherClientBuilder builder) {
+        return builder.buildEventGridEventPublisherClient();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnProperty(value = "spring.cloud.azure.eventgrid.event-schema", havingValue = "EVENT_GRID_EVENT", matchIfMissing = true)
+    EventGridPublisherAsyncClient<EventGridEvent> eventGridEventPublisherAsyncClient(EventGridPublisherClientBuilder builder) {
+        return builder.buildEventGridEventPublisherAsyncClient();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnProperty(value = "spring.cloud.azure.eventgrid.event-schema", havingValue = "CLOUD_EVENT")
+    EventGridPublisherClient<CloudEvent> eventGridCloudEventPublisherClient(EventGridPublisherClientBuilder builder) {
+        return builder.buildCloudEventPublisherClient();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnProperty(value = "spring.cloud.azure.eventgrid.event-schema", havingValue = "CLOUD_EVENT")
+    EventGridPublisherAsyncClient<CloudEvent> eventGridCloudEventPublisherAsyncClient(EventGridPublisherClientBuilder builder) {
+        return builder.buildCloudEventPublisherAsyncClient();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnProperty(value = "spring.cloud.azure.eventgrid.event-schema", havingValue = "CUSTOM_EVENT")
+    EventGridPublisherClient<BinaryData> eventGridCustomEventPublisherClient(EventGridPublisherClientBuilder builder) {
+        return builder.buildCustomEventPublisherClient();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    @ConditionalOnProperty(value = "spring.cloud.azure.eventgrid.event-schema", havingValue = "CUSTOM_EVENT")
+    EventGridPublisherAsyncClient<BinaryData> eventGridCustomEventPublisherAsyncClient(EventGridPublisherClientBuilder builder) {
+        return builder.buildCustomEventPublisherAsyncClient();
+    }
```

Comment:
Shall we support multiple events schema publisher clients in one application? otherwise, it's required to publish these events through different applications.

Reply:
That could be a feature, but let's not rush in this version.